pilot/pkg/serviceregistry/kube/controller.go

```diff
@@ -423,22 +428,31 @@ func (c *Controller) GetProxyServiceInstances(proxy *model.Proxy) ([]*model.Serv
 			continue
 		}
 
+		endpoints := &endpointsForPodInSameNS
+		if ep.Namespace != proxyNS {
```

Comment:
this should never occur

Reply:
I think the naming is a little misleading, probably I should change the `proxyNS` to something like `podNS`.
But basically, I'm trying to put the [k8s endpoint](https://kubernetes.io/docs/reference/generated/kubernetes-api/v1.10/#endpoints-v1-core) which has a different namespace compared to the referred pod to have lower priority when Pilot is generating listeners.

Also note it's possible if you define a [k8s service](https://kubernetes.io/docs/reference/generated/kubernetes-api/v1.10/#service-v1-core) but doesn't specify labels, and then define a [k8s endpoint](https://kubernetes.io/docs/reference/generated/kubernetes-api/v1.10/#endpoints-v1-core) that has the same name as the `k8s service`, in the `k8s endpoint` you can use IP address to bind it with pod, even if it's not in the endpoint's namespace.

Reply:
@yangminzhu This is not a right way to use endpoint and svc. If we do that we can not access the service by cluster ip(
```
root@szvp000201060:~# iptables-save |grep  10.0.0.133 
-A KUBE-SERVICES -d 10.0.0.133/32 -p tcp -m comment --comment "default/reviews:http has no endpoints" -m tcp --dport 9080 -j REJECT --reject-with icmp-port-unreachable

root@szvp000201060:~# kubectl get ep -n istio-system reviews
NAME      ENDPOINTS                                           AGE
reviews   172.17.0.18:9080,172.17.0.19:9080,172.17.0.5:9080   2m43s
```
).
Also we can not access it in istio mesh, I have tested this with bookinfo example.

Reply:
Could you be more specific what do you want for this comment?
For my previous comment, It's just a way (I'm not saying it's a right way or not) to use endpoint that is allowed by k8s, you would be still able to access the service by its name directly.

Reply:
I mean this `if ep.Namespace != proxyNS {` should never happen. If it does, then the svc can not be accessed.

Reply:
So, I think we do not need these code in this section.